### DISCUSSION TITLE
How do I get execute permissions on a Fabric Workload Item?

### DISCUSSION THREAD

**Discussion by q-benwillis** [microsoft/Microsoft-Fabric-workload-development-sample#119] [Help] [answered]
When I create a new workload item and call the [resolve permissions endpoint](https://learn.microsoft.com/en-us/rest/api/fabric/workload/workloadcontrolapi/workload-control/resolve-item-permissions?tabs=HTTP#itempermissions) I only have "Read", "Write" and "Reshare" permissions. How can I also get the execute permission?

**Comment by michal-mano** [comment]
Hello,
Can you try adding the "Power BI Service Item.Execute.All" delegated permission to your Entra ID App?
![image](https://github.com/user-attachments/assets/f07a8270-b8b1-467a-9d78-f1dd6596ed7a)
After adding this, you'll have to provide consent as the user for this additional permission.

**Reply by q-benwillis** [reply]
It's not that I want my entra id app to have the execute permission, it's that I would have though the user should have execute permission on fabric workload items that they create.

There's an example here in the sample repo of using the resolve permissions endpoint to check the user has the right permissions to carry out an action: https://github.com/microsoft/Microsoft-Fabric-workload-development-sample/blob/b4cd86d3346b55d5cb5fa5af780f82ea4418b6d1/Backend/src/Services/AuthorizationHandler.cs#L37-L60

Ideally I want to check the user has the "execute" permission but I don't know if they can actually get that permission?

**Reply by michal-mano** [reply, marked as answer]
Okay, understood, thanks for clarifying.
We do not yet support custom permissions such as Execute. All items are generated with just Read, Write, Reshare
